rct229/rulesets/ashrae9012019/section11/section11rule1.py

```diff
@@ -0,0 +1,126 @@
+            "a service water-heating system shall be modeled that matches the system type in the baseline building design",
+            ruleset_section_title="Service Water Heating",
+            standard_section="Table G3.1 #11, proposed column, a & b",
+            is_primary_rule=True,
+            list_path="ruleset_model_descriptions[0]",
+        )
+
+    class RMDRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section11Rule1.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=True, BASELINE_0=True, PROPOSED=True
+                ),
+                each_rule=Section11Rule1.RMDRule.SWHDistributionRule(),
+                index_rmd=BASELINE_0,
+                list_path="$.service_water_heating_distribution_systems[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_u = context.USER
+            rmd_b = context.BASELINE_0
+            rmd_p = context.PROPOSED
+
+            swh_and_equip_dict_u = (
+                get_swh_equipment_associated_with_each_swh_distribution_system(rmd_u)
+            )
+            swh_and_equip_dict_b = (
+                get_swh_equipment_associated_with_each_swh_distribution_system(rmd_b)
+            )
+            swh_and_equip_dict_p = (
+                get_swh_equipment_associated_with_each_swh_distribution_system(rmd_p)
+            )
+
+            errors = []
+            user_proposed_comparison = []
+            user_baseline_comparison = []
+            for swh_dist_id_u in find_all(
+                "$.service_water_heating_distribution_systems[*].id", rmd_u
+            ):
+                if swh_and_equip_dict_u[swh_dist_id_u].uses:
+                    swh_use_u = find_one(
+                        f'$.buildings[*].building_segments[*].zones[*].spaces[*].service_water_heating_uses[*][?(@.served_by_distribution_system="{swh_dist_id_u}")]',
+                        rmd_u,
+                    )
+                    if swh_use_u is not None and swh_use_u.get("use") > 0.0:
+                        user_proposed_comparison = (
+                            compare_swh_dist_systems_and_components(
+                                rmd1=rmd_p,
+                                rmd2=rmd_u,
+                                compare_context_str="AppG 11-1 P_RMD Equals U_RMD",
+                                swh_distribution_id=swh_dist_id_u,
+                            )
+                        )
+                else:
+                    user_baseline_comparison = compare_swh_dist_systems_and_components(
+                        rmd1=rmd_p,
+                        rmd2=rmd_b,
+                        compare_context_str="AppG 11-1 P_RMD Equals B_RMD",
+                        swh_distribution_id=swh_dist_id_u,
+                    )
```

Comment:
This logic is when there is distribution system defined but no swh_use load, so we only compare proposed and baseline (typo on the parameter name).

However, one more logic need to add outside of the for loop, which is if there is no distribution system in the user model, then we need to compare proposed and baseline. You will need to start a new loop based on proposed model (if user model has no distribution system) and run through each swh distribution system for comparison.

Reply:
Good point. Could you see if my updates align with your explanation?